tests/Doctrine/Tests/Common/Persistence/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -159,6 +159,35 @@ public function testWillIgnoreCacheEntriesThatAreNotMetadataInstances()
 
         self::assertSame($metadata, $this->cmf->getMetadataFor('Foo'));
     }
+
+    public function testFallbackMetadataShouldBeCached()
+    {
+        /** @var Cache|PHPUnit_Framework_MockObject_MockObject $cacheDriver */
+        $cacheDriver = $this->createMock(Cache::class);
+        $cacheDriver->expects($this->once())->method('save');
+
+        $this->cmf->setCacheDriver($cacheDriver);
+
+        $classMetadata = $this->createMock(ClassMetadata::class);
+
+        $this->cmf->fallbackCallback = static function () use ($classMetadata) {
+            return $classMetadata;
+        };
+
+        $this->cmf->getMetadataFor('Foo');
+    }
+
+    public function testSetMetadataForShouldUpdateCache()
+    {
+        /** @var Cache|PHPUnit_Framework_MockObject_MockObject $cacheDriver */
```

Comment:
We should rather rely on `PHPUnit\Framework\MockObject\MockObject` since it's part of PHPUnit's forward compatibility layer (even though the rest of the test uses the old interface).